ui/tsconfig.json

```diff
@@ -13,6 +13,7 @@
     "useDefineForClassFields": false,
     "sourceMap": true,
     "baseUrl": ".",
+    "downlevelIteration": true,
```

Comment:
Very good question. The e2e tests wouldn't run without this because of `[...dataset]` I think. I spent a bit of time on it but not enough to really understand the issue.